src/charm.py

```diff
@@ -55,13 +68,27 @@ def __init__(self, *args):
             self.state, self.workload, SUBSTRATE
         )
 
+        # --- UPGRADES ---
+        try:
+            self.refresh = charm_refresh.Machines(
+                MachinesEtcdRefresh(workload_name="etcd", charm_name="charmed-etcd", charm=self)
```

Comment:
```suggestion
                MachinesEtcdRefresh(workload_name="etcd", charm_name="charmed-etcd", _charm=self)
```
nit; to avoid potential clashes with future args in the public API

Reply:
Understand, but will keep it for now.